core/src/main/java/hudson/slaves/WorkspaceList.java

```diff
@@ -112,7 +112,9 @@ public String toString() {
         public final @Nonnull FilePath path;
 
         protected Lease(@Nonnull FilePath path) {
-            path.getRemote(); // null check
+            if (path == null) { // Protection from old API
+                throw new NullPointerException("The specified FilePath is null");
```

Comment:
Personally not totally convinced that the `NPE` is the correct way to manage this case. In the core, there are lots more `IAE` (> 40) than `NPE` (~8) for such cases.

More generally speaking, there is a debate concerning the fact that only the JVM should throw `NPE` but seems that `Effective Java` says the contrary.

So -0.1 from me